ghost/members-api/lib/repositories/MemberRepository.js

```diff
@@ -284,13 +284,23 @@ module.exports = class MemberRepository {
 
         //checks for custom signUp forms
         if (memberData.newsletters && memberData.newsletters.length > 0) {
-            const savedNewsletter = await this._newslettersService.browse({filter: `id:'${memberData.newsletters[0].id}'`});
-            if (savedNewsletter.length === 0) {
-                throw new errors.BadRequestError({message: tpl(messages.invalidNewsletterId, {id: memberData.newsletters[0].id})});
-            }
-            if (savedNewsletter[0].status === 'archived') {
-                memberData.newsletters = [];
+            const memberNewsletterIds = memberData.newsletters.map(newsletter => newsletter.id);
+            const validNewsletters = await this._newslettersService.browse({
```

Comment:
technically we don't know that the retrieved newsletters are `valid` yet, how about just `newsletters`?